pkg/log/log.go

```diff
@@ -46,3 +52,22 @@ func SetLogger(l logr.Logger) {
 // to another logr.Logger.  You *must* call SetLogger to
 // get any actual logging.
 var Log = NewDelegatingLogger(NullLogger{})
+
+// FromContext returns a logger with predefined values from a context.Context.
+func FromContext(ctx context.Context, keysAndValues ...interface{}) logr.Logger {
+	var log logr.Logger
+	if ctx == nil {
+		log = Log
+	} else {
+		lv := ctx.Value(contextKey)
+		log = lv.(logr.Logger)
+	}
+	log.WithValues(keysAndValues...)
+	return log
+}
+
+// SetContext takes a context and sets the logger as one of its keys.
+// Use FromContext function to retrieve the logger.
+func SetContext(ctx context.Context, log logr.Logger) context.Context {
```

Comment:
Seeing this in use without knowing in advance that it is just adding the logger to a given context is a bit confusing. At first I thought this was setting package level variable the same way that `SetLogger` does:
```suggestion
func ContextWithLogger(ctx context.Context, log logr.Logger) context.Context {
```

Reply:
Since at the end the function call will be prefixed with the package name, what about `IntoContext`? That would mean the call is `log.IntoContext(ctx, log)`

Reply:
+1 either way